gradle-plugin/src/main/java/com/contrastsecurity/gradle/plugin/ResolveAgentTask.java

```diff
@@ -0,0 +1,96 @@
+package com.contrastsecurity.gradle.plugin;
+
+import static java.util.Objects.requireNonNull;
+
+import com.contrastsecurity.gradle.plugin.extensions.ContrastConfigurationExtension;
+import com.contrastsecurity.models.AgentType;
+import com.contrastsecurity.sdk.ContrastSDK;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import javax.inject.Inject;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+/**
+ * Task for setting the agent in the project. This task takes in a user provided agent from the
+ * {@link ContrastConfigurationExtension} and writes it to a file in the project's build directory.
+ * If no agent is provided, then this task downloads an agent with the {@link ContrastSDK} and puts
+ * it in the projects build directory.
+ */
+public abstract class ResolveAgentTask extends DefaultTask {
+
+  private final ContrastSDK sdk;
+
+  @Inject
+  public ResolveAgentTask(final ContrastSDK sdk) {
+    super();
+    requireNonNull(sdk);
+    this.sdk = sdk;
+  }
+
+  /** The agent path provided via the {@link ContrastConfigurationExtension} */
+  @Internal
+  public abstract Property<String> getProvidedPath();
+
+  /** The agent returned by the task */
+  @OutputFile
+  public abstract RegularFileProperty getAgent();
+
+  @TaskAction
+  public void resolveAgent() throws IOException {
+
+    final Logger logger = getProject().getLogger();
+
+    final ContrastConfigurationExtension extension =
+        getProject().getExtensions().getByType(ContrastConfigurationExtension.class);
+    requireNonNull(extension);
+
+    final Path agentBuildPath = getAgent().getAsFile().get().toPath();
+
+    // Check if an agent was provided, and if so write that to the expected location
+    if (getProvidedPath().isPresent()) {
+      final String providedAgent = getProvidedPath().get();
+      Path providedPath = Paths.get(providedAgent);
+      if (!Files.exists(providedPath)) {
+        logger.debug("Unable to find java agentPath at " + providedAgent);
+
+      } else {
+        logger.debug("Agent provided via configuration retrieved");
+        writeToAgent(Files.readAllBytes(providedPath), agentBuildPath);
```

Comment:
Do you want to return after this?  Right now it's going and still getting it from Team Server and calling writeToAgent again.